### PR TITLE
forgot password flow sync with notification service

### PR DESCRIPTION
**Pull request by SunmadeBabafemi**
1. a user forgets his password and proceeds to the forgot password page
2. he enters his registered email to be sent a reset link embedded with the reset token
3. the email is verified to exist on the db and then the link is sent
4. clicking the link brings him to the password reset page where he enters his new password
5. the password is checked as against the old password to ensure they are not the same
6. also the token is verified to ensure it is the same user.. after which the password is changed successfully

### CODE REVIEW

**Review by Chimexx — COMMENTED** [review]
You have written some of this code in your previous PR, including the installed rabbitmq  packages, I also made some commits in that PR. I don't know what effect this may have. So I will advice we keep this pending till your previous PR is Merged.

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -118,4 +120,45 @@ export class AuthService {
...
+    username: user.fullName
+    }
+
+    const success = await this.emailVerificationService.sendResetPasswordLink(emailPayload)
+
+    return  [user.id, success]
+  }
+
+  async resetPassword(authorizationToken: string, body: PasswordResetDto): Promise<User>{
+    const token = authorizationToken.split(' ')[1]
```
[the remark this answers is not in the document]
Comment (SunmadeBabafemi) [reply src/auth/auth.service.ts:148]: it was messing with the code before.. that's why i removed it... i'll look at it again

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -118,4 +120,45 @@ export class AuthService {
   async getNewTokens(refreshToken: string) {
     return await this.jwtHelperService.getNewTokens(refreshToken);
   }
+
+  async forgotPassword(email: string): Promise<string[]>{
+    const user: User = await this.userRepo.findOne({where: {email}})
```
[the remark this answers is not in the document]
Comment (SunmadeBabafemi) [reply src/auth/auth.service.ts:125]: alright

**Comment by SunmadeBabafemi** [comment]
> You have written some of this code in your previous PR, including the installed rabbitmq packages, I also made some commits in that PR. I don't know what effect this may have. So I will advice we keep this pending till your previous PR is Merged.

or should i just do a git pull to pull all the changes in that branch into this and re-initiate the forgot password flow here so as to avoid any conflict

**Comment by Chimexx** [comment]
Previous PR has been merged, fix your conflicts

**Pull request closed without merging**